pandas/tests/indexes/common.py

```diff
@@ -937,3 +937,14 @@ def test_empty(self):
         index = self.create_index()
         assert not index.empty
         assert index[:0].empty
+
+    @pytest.mark.parametrize('how', ['outer', 'inner', 'left', 'right'])
+    def test_join_self(self, how):
+        index = self.create_index()
+        joined = index.join(index, how=how)
+
+        if index is index.unique():
```

Comment:
it's only occuring for right joins:

```
(Pdb) index.join(index) is index
True
(Pdb) index.join(index, how='right') is index
False
(Pdb) index
TimedeltaIndex(['0 days 01:00:00', '1 days 01:00:00', '2 days 01:00:00',
                '3 days 01:00:00', '4 days 01:00:00'],
               dtype='timedelta64[ns]', freq='D')

(Pdb) index.copy() is index
False
(Pdb) index = pd.Index([1,2,3])
(Pdb) index.join(index, how='right')
Int64Index([1, 2, 3], dtype='int64')
(Pdb) index.join(index, how='right') is index
True
(Pdb) index is index.copy()
False

```

Are we are making a copy somewhere in the right-join path?